plugin/openassetio_manager_bal/bal.py

```diff
@@ -62,12 +65,18 @@ def load_library(path: str) -> dict:
     """
     Loads a library from the supplied path
     """
-    if not path:
-        # Allow an empty path, meaning an empty library.
-        return {}
+    library = {"vars": {"bal_library_path": "", "bal_library_dir": ""}}
```

Comment:
[question] Why a dict of dicts here, and not just?
```python
library_vars = {"bal_library_path": "", "bal_library_dir": ""}
```

Reply:
In writing the answer, I realised that the current implementation already actually fully supported arbitrary variable loading from the library. I was always thinking we'd add that in the future, but was keeping it out of scope for now.  So I took to opportunity to maybe make it slightly more readable, and formalize it [here](https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/pull/23/commits/b16858eb2f9b3e857e38ffb042d96fdc2e7c20da).